core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1318,7 +1318,7 @@ abstract class RDD[T: ClassTag](
   /**
    * Save this RDD as a SequenceFile of serialized objects.
    */
-  def saveAsObjectFile(path: String) {
+  def saveAsObjectFile(path: String, codec: Option[Class[_ <: CompressionCodec]] = None) {
```

Comment:
I think we'll have to see if this technically breaks binary compatibility though; it may need a separate method. MiMa will check.

Also I imagine it would be nice to give Python the same parallel treatment.